### PR TITLE
add node, link water quality getters

### LINKED ISSUE #206: Node and link water quality features
Build out the node and link access to pollutant/water quality values. I am actively working on this.

### PR DESCRIPTION
**Pull request by katmratliff**
- adds getters for subcatchment `pollut_quality` and `runoff_total_loading`
- adds getters for link `pollut_quality` and `total_loading`
- adds getters for node `pollut_quality`
- updates tests

closes #206

### CODE REVIEW

**Comment by katmratliff** [comment]
@bemcdonnell @jennwuu the linux, mac, and x64 libs still need to be updated

**Comment by jennwuu** [comment]
@katmratliff I looked through the pyswmm code and it looks awesome. The reason why I have not approved is because I'm wondering if the OWA SWMM PR should be approved first so we can reference the commit # in pyswmm. @bemcdonnell thoughts?

**Comment by katmratliff** [comment]
@jennwuu I agree that the SWMM PR should be approved first - just wanted to get the ball rolling over here too :)

**Comment by katmratliff** [comment]
> edit: actually I just noticed there is a null pointer error which happens in 64 bit environment on appveyor.

@jennwuu @bemcdonnell are you all familiar with why this might happen? I haven't used the 64 bit environment, so I'm not sure why that's throwing an error.

**Comment by jennwuu** [comment]
@katmratliff I'm reading through SWMM and pyswmm code and I have some questions:

1.  Sorry for silly questions, can you explain why a double pointer for double is used instead of a single pointer for double? Is it because a pointer is used to hold the array of double, and then another pointer to retrieve the first pointer?

2. How come cast function instead? Can you try .contents of the pointer instead? Similar to https://github.com/OpenWaterAnalytics/pyswmm/blob/028fc9a2c731bed4129ee100e295900d989867fd/pyswmm/swmm5.py#L1812-L1815

I think there is something weird with casting function for pointers when you use 32 bit versus 64 bit of python.

**Comment by katmratliff** [comment]
@jennwuu cast is also used in the rain gage API and I have used it previously for the subcatchment pollutant API, and that hasn't seemed to cause problems in the past, right?

**Comment by jennwuu** [comment]
@katmratliff oh! After I replaced swmm5-x64.dll from appveyor build, I don't get the null pointer error anymore. 

Try replacing the swmm5-x64.dll from https://ci.appveyor.com/project/OpenWaterAnalytics/stormwater-management-model/builds/28128188/job/cmqnotdgovhhkb8n/artifacts again.

**Comment by katmratliff** [comment]
@jennwuu hooray! that makes so much sense :)

**Comment by katmratliff** [comment]
@jennwuu now it looks like it's failing because of the same reasons that the [SWMM PR](https://github.com/OpenWaterAnalytics/Stormwater-Management-Model/pull/262) is failing, compiler differences.

**Comment by katmratliff** [comment]
@jennwuu does Travis CI save the build artifacts from OWA-SWMM anywhere? it looks like they get built and zipped up, but I can't find them!

**Comment by katmratliff** [comment]
@bemcdonnell @jennwuu linux and both windows builds are passing now (I had to relax test precision some). I'm not sure what's going on with the mac build; there's a null pointer access error there.

**Comment by jennwuu** [comment]
@katmratliff the travis CI build artifacts get uploaded as draft releases under the OWA-SWMM release page. Do you have access to the draft releases? Sorry for the late reply!

**Comment by katmratliff** [comment]
@jennwuu thanks! I do have access and updated the mac lib. we'll see if the mac build passes the tests now...

**Comment by jennwuu** [comment]
@katmratliff looks like the mac build is failing on numerical difference now

File: pyswmm/__init__.py [hunk 1]
```diff
@@ -20,7 +20,7 @@
 VERSION_INFO = (0, 6, 1, 'dev0')
 __version__ = '.'.join(map(str, VERSION_INFO))
 __swmm_version__ = '5.2.1'
-__swmm_sha__ = '854df626b32f74e5cfb3a07fbea3f51b5d7855bb'
```
Comment (jennwuu) [inline pyswmm/__init__.py]: I think this commit is failing on OWA-SWMM because it's missing signature from clahub.

Comment (katmratliff) [reply pyswmm/__init__.py]: @jennwuu it's strange, it looks like it's failing b/c I haven't signed (although I have), but then when you go to the PR, all the checks have passed:
image
Not sure what's going on there!

Comment (jennwuu) [reply pyswmm/__init__.py]: @bemcdonnell thoughts on why the commit 854df6 seems to be failing on OWA-SWMM?

Comment (katmratliff) [reply pyswmm/__init__.py]: @bemcdonnell @jennwuu I'm not sure why clahub shows as failing for SWMM -- I have signed it.  Any thoughts?

Comment (bemcdonnell) [reply pyswmm/__init__.py]: This PR looks good to me? All the checks have passed. Hmm

Comment (katmratliff) [reply pyswmm/__init__.py]: @bemcdonnell for some reason clahub is showing as failing in my most recent SWMM commits. But I've signed it so I'm not sure why... I believe that's what held this pyswmm PR up previously, the fact that the OWA-SWMM commit referenced in this version of pyswmm shows as failing (correct me if I'm wrong @jennwuu)

Comment (bemcdonnell) [reply pyswmm/__init__.py]: Oh sorry, I didn't notice you were talking about SWMM. let me look

Comment (jennwuu) [reply pyswmm/__init__.py]: @bemcdonnell yeah, it shows failed on swmm due missing clahub. really odd, not sure why? Maybe we should try to rebuild this commit?

Comment (jennwuu) [reply pyswmm/__init__.py]: @katmratliff https://www.clahub.com/agreements/OpenWaterAnalytics/Stormwater-Management-Model just to double check, this shows signed for you right?

Comment (katmratliff) [reply pyswmm/__init__.py]: @jennwuu yes it shows as signed for me

Comment (jennwuu) [reply pyswmm/__init__.py]: @bemcdonnell can we bump swmm version to 5.3.0 from 5.3.0.dev0 for SWMM and that can be the new commit # referenced in pyswmm?

Comment (jennwuu) [reply pyswmm/__init__.py]: @bemcdonnell  - poke -

Comment (jennwuu) [reply pyswmm/__init__.py]: @katmratliff, I’m happy with this PR once the OWA SWMM version gets updated. :D

**Comment by katmratliff** [comment]
@bemcdonnell @jennwuu I got rid of `brew install gcc` in the Travis dependencies because that was causing Travis to fail and it doesn't seem necessary for running our tests. Does that make sense to you all?

**Comment by jennwuu** [comment]
@katmratliff I don't think we need gcc unless we are using this section of pyswmm https://github.com/OpenWaterAnalytics/pyswmm/tree/master/src @bemcdonnell

**Comment by katmratliff** [comment]
@bemcdonnell are you ok with including example files in the documentation? if you or @jennwuu have any others that would be useful please send them my way and I will include them as well

**Comment by katmratliff** [comment]
@bemcdonnell I'm going to wait for your stamp of approval before merging this PR :)

**Pull request merged**